### DISCUSSION TITLE
Incremental migration to 2.0

### DISCUSSION THREAD

**Discussion by michaelbukachi** [sqlalchemy/sqlalchemy#9310] [Usage Questions] [answered]
Are there any strategies for migrating to SQLAlchemy 2.0 for a huge codebase? Especially with types involved?

**Comment by zzzeek** [comment, marked as answer]
sure, you would go through the steps in the migration guide at https://docs.sqlalchemy.org/en/20/changelog/migration_20.html so that you run without any warnings on 1.4 first.   Then you can start testing against 2.0, then once your codebase is on 2.0 you can begin migrating the declarative mappings using the guidelines at https://docs.sqlalchemy.org/en/20/changelog/whatsnew_20.html#migrating-an-existing-mapping .

to the degree that "types are involved" it depends on if you are dependent on third party stubs or the SQLAlchemy 1.4 stubs + mypy plugin.     To run under 1.4 or 2.0 using legacy declarative mappings with typing, you need to use the mypy plugin in the interim if you want those mappings to be typed.
